src/main/java/com/redhat/rhjmc/containerjfr/net/web/WebServer.java

```diff
@@ -147,6 +152,7 @@
             TargetConnectionManager targetConnectionManager,
             Logger logger) {
         this.server = server;
+        this.client = client;
```

Comment:
I think this is probably okay, but what do you think about injecting a `Provider<CloseableHttpClient>`? Then later on when you need to use a client, you can create, use, and clean up the client as needed. Right now there will be a single client sticking around for the life of the webserver. That's not likely a real problem, but I guess just a stylistic thing.